File: aroundHubSpringBoot/src/main/java/com/example/aroungHubJavaSpringBoot/aroundHubSpringBoot/service/impl/ProductServiceImpl.java
package com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.service.impl;

import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.dto.ProductDto;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.entity.ProductEntity;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.handler.ProductDataHandler;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.service.ProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ProductServiceImpl implements ProductService {

    ProductDataHandler productDataHandler;

    @Autowired
    public ProductServiceImpl(ProductDataHandler productDataHandler){
        this.productDataHandler = productDataHandler;
    }


    @Override
    public ProductDto saveProduct(String productId, String productName, int productPrice, int productStock) {
        ProductEntity productEntity = productDataHandler.saveProductEntity(productId, productName, productPrice, productStock);

        ProductDto productDto = new ProductDto(productEntity.getProductId(), productEntity.getProductName(), productEntity.getProductPrice(), productEntity.getProductStock());

        return productDto;
    }

    @Override
    public ProductDto getProduct(String productId) {
        ProductEntity productEntity = productDataHandler.getProductEntity(productId);

        ProductDto productDto = new ProductDto(productEntity.getProductId(), productEntity.getProductName(), productEntity.getProductPrice(), productEntity.getProductStock());

        return productDto;
    }
}

File: aroundHubKotlinSpringBoot/src/main/kotlin/com/example/aroungHubKotlinSpringBoot/aroundHubKotlinSpringBoot/service/impl/ProductServiceImpl.kt
package com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.service.impl

import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.dto.ProductDto
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.entity.ProductEntity
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.handler.ProductDataHandler
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.service.ProductService
import org.springframework.stereotype.Service

@Service
class ProductServiceImpl(
    val productDataHandler : ProductDataHandler
): ProductService {
    override fun saveProduct(
        productId: String,
        productName: String,
        productPrice: Int,
        productStock: Int
    ): ProductDto {
        var productEntity : ProductEntity = productDataHandler.saveProductEntity(productId, productName, productPrice, productStock)

        var productDto : ProductDto = ProductDto(productEntity.productId, productEntity.productName, productEntity.productPrice, productEntity.productStock)
        return productDto
    }

    override fun getProduct(productId: String): ProductDto {
        var productEntity: ProductEntity = productDataHandler.getProductEntity(productId)

        var productDto: ProductDto = ProductDto(productEntity.productId, productEntity.productName, productEntity.productPrice, productEntity.productStock)

        return productDto
    }
}